packages/smooth_app/lib/pages/product/edit_ingredients_page.dart

```diff
@@ -0,0 +1,339 @@
+    widget.product.ingredientsText = ingredientsText;
+    final Status status =
+        await OpenFoodAPIClient.saveProduct(user, widget.product);
+    if (status.error != null) {
+      throw Exception("Couldn't save the product. ${status.error}");
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeProvider themeProvider = context.watch<ThemeProvider>();
+    final ThemeData darkTheme = SmoothTheme.getThemeData(
+      Brightness.dark,
+      themeProvider.colorTag,
+    );
+
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+
+    return Scaffold(
+      extendBodyBehindAppBar: true,
+      appBar: AppBar(
+        title: Text(appLocalizations.ingredients_editing_title),
+        backgroundColor: Colors.transparent,
+        flexibleSpace: ClipRect(
+          child: BackdropFilter(
+            filter: ImageFilter.blur(sigmaX: 5.0, sigmaY: 5.0),
+            child: Container(
+              color: Colors.transparent,
+            ),
+          ),
+        ),
+      ),
+      body: Stack(
+        children: <Widget>[
+          if (_imageProvider != null)
+            ConstrainedBox(
+              constraints: const BoxConstraints.expand(),
+              child: Image(
+                image: _imageProvider!,
+                fit: BoxFit.cover,
+              ),
+            ),
+          if (_imageProvider == null && widget.imageIngredientsUrl != null)
+            ConstrainedBox(
+              constraints: const BoxConstraints.expand(),
+              child: Image(
+                fit: BoxFit.cover,
+                image: NetworkImage(widget.imageIngredientsUrl!),
+              ),
+            ),
+          if (_imageProvider == null && widget.imageIngredientsUrl == null)
+            Container(color: Colors.white),
+          if (_updatingImage)
+            const Center(
+              child: CircularProgressIndicator(),
+            ),
+          if (!_updatingImage)
+            Align(
+              alignment: Alignment.bottomLeft,
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 16.0),
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: <Widget>[
+                    Flexible(
+                      flex: 1,
+                      child: Align(
+                        alignment: Alignment.bottomRight,
+                        child: Padding(
+                          padding: const EdgeInsets.only(bottom: 16.0),
+                          child: _ActionButtons(
+                            getImage: _onTapGetImage,
+                            hasImage: widget.imageIngredientsUrl != null,
+                          ),
+                        ),
+                      ),
+                    ),
+                    Flexible(
+                      flex: 1,
+                      child: Container(
+                        height: 400.0,
```

Comment:
Where does `400` come from? Please add a `TODO` if it's a temporary guess or use a constant from `design_constants.dart`.

Reply:
Whoops, it's irrelevant now, it's sized by the Flexible above.  Will remove it.